### DISCUSSION TITLE
esptool write_flash --erase-all option corrupting flash

### DISCUSSION THREAD

**Discussion by git-n-pissed** [micropython/micropython#10174] [General]
Not sure if this is a regression caused by changes to esptool or MP, but back on esptool v2.6, the `write_flash --erase-all` option worked as expected, erasing the flash before writing, all in one command.  I just upgraded to esptool v4.4 and this option now causes the flash to be corrupted on write without any indication of failure until you REPL in.  What's worse, if you also use the `--verify` option, you are told the `--verify` option is deprecated and that flash contents are always read back after flashing, but verification goes ahead anyway, and eventually fails.  I'm seeing the same behavior on ESP8266, ESP32-WROOM32, and ESP32-S3-WROOM32 all using a stock downloaded build of MP (v1.19.1).

```
python3 -m esptool --baud 921600 --chip esp32s3 --port /dev/ttyS9 write_flash --erase-all --verify 0 /mnt/c/Users/git-n-pissed/Desktop/GENERIC_S3-20220618-v1.19.1.bin
esptool.py v4.4
Serial port /dev/ttyS9
Connecting....
Chip is ESP32-S3 (revision v0.1)
Features: WiFi, BLE
Crystal is 40MHz
MAC: 00:00:00:00:00:00
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Erasing flash (this may take a while)...
Chip erase completed successfully in 20.4s
Compressed 1387904 bytes to 916115...
Wrote 1387904 bytes (916115 compressed) at 0x00000000 in 19.8 seconds (effective 560.5 kbit/s)...
Hash of data verified.

Leaving...
Verifying just-written flash...
(This option is deprecated, flash contents are now always read back after flashing.)
Verifying 0x157670 (1406576) bytes @ 0x00000000 in flash against /mnt/c/Users/git-n-pissed/Desktop/GENERIC_S3-20220618-v1.19.1.bin...
-- verify FAILED (digest mismatch)
```

**Comment by jimmo** [comment]
@git-n-pissed It seems unlikely that if this was working for you before and all that changed was esptool that this would be a MicroPython issue. Maybe better follow up with esptool -- https://github.com/espressif/esptool

Have you tried some other versions between 2.6 and 4.4 ?

**Reply by git-n-pissed** [reply]
> @git-n-pissed It seems unlikely that if this was working for you before and all that changed was esptool that this would be a MicroPython issue. Maybe better follow up with esptool -- https://github.com/espressif/esptool
> 
> Have you tried some other versions between 2.6 and 4.4 ?

I agree that it does seem more likely that this is an esptool problem, but I figured I'd post here first since I'm exclusively using MP on these boards and some other ROMs require different address parameters when flashing.  For instance, some ROMS require that the start address be 0x100 when flashing an ESP32, that is the only reason I though this might be related to MP some how, since MP has a start address of 0 when flashing.

I have not tried other versions of esptool, so I don't know where exactly the break occured.

**Reply by git-n-pissed** [reply]
Issue created: https://github.com/espressif/esptool/issues/805
